local-volume/utils/update-pv-to-beta/README.md

```diff
@@ -0,0 +1,99 @@
+# local-volume-update-pv-to-beta
+
+local-volume-update-pv-to-beta is used to update local PVs alpha node affinity annotation to beta
+Below is how to compile and use the tool.
+
+## Deployment
+
+### Compile the provisioner
+``` console
+make
+```
+
+### Make the container image and push to the registry
+``` console
+make push
+```
+
+### Clean the binary
+``` console
+make clean
+```
+
+### Create local PV with alpha node affinity
+below is an example of local PV
+``` pv.yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: example-local-pv-1
+  annotations:
+    "volume.alpha.kubernetes.io/node-affinity": '{
+      "requiredDuringSchedulingIgnoredDuringExecution": {
+        "nodeSelectorTerms": [
+          { "matchExpressions": [
+            { "key": "kubernetes.io/hostname",
+              "operator": "In",
+              "values": ["nickren-14"]
+            }
+          ]}
+         ]}
+        }'
+spec:
+  capacity:
+    storage: 200Mi
+  accessModes:
+  - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: local-storage
+  local:
+    path: /mnt/disks/vol/vol1
+```
+create PV and describe it
+``` kubectl describe pv example-local-pv-1
+Name:            example-local-pv-1
+Labels:          <none>
+Annotations:     volume.alpha.kubernetes.io/node-affinity={ "requiredDuringSchedulingIgnoredDuringExecution": { "nodeSelectorTerms": [ { "matchExpressions": [ { "key": "kubernetes.io/hostname", "operator": "In", "valu...
+Finalizers:      [kubernetes.io/pv-protection]
+StorageClass:    local-storage
+Status:          Available
+Claim:
+Reclaim Policy:  Retain
+Access Modes:    RWO
+Capacity:        200Mi
+Node Affinity:   <none>
+Message:
+Source:
+    Type:  LocalVolume (a persistent volume backed by local storage on a node)
+    Path:  /mnt/disks/vol/vol1
+Events:    <none>
+```
+
+### Create ServiceAccount and kubernetes job to update local PV alpha node affinity to beta
+``` console
+kubectl create -f deployment/kubernetes/admin-account.yaml
```

Comment:
Should the service account be removed after the job is complete?

Reply:
done